code/modules/mob/living/carbon/xenomorph/abilities/royal/royal_powers.dm

```diff
@@ -0,0 +1,219 @@
+
+	else if (curr_effect_type == ROYAL_SCREECH_DEBUFF)
+		playsound(xeno.loc, screech_sound_effectt, 55, 0, status = 0)
+		xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] emits a guttural roar!"))
+		xeno.create_shriekwave(color = "#FF0000")
+		var/slow_duration = 40
+
+		for(var/mob/living/carbon/human/human in view(3, xeno))
+			human.visible_message(SPAN_DANGER("[xeno]'s roar shakes your entire body, causing you to fall over in pain!"))
+			if (!(xeno.screech_status_flags & ROYAL_SCREECH_DEBUFF))
+				shake_camera(human, 2, 3)
+				human.Daze(debuff_daze)
+				human.KnockDown(get_xeno_stun_duration(human, 0.5))
+				new /datum/effects/xeno_slow(human, xeno, null, null, get_xeno_stun_duration(human, slow_duration))
+
+	apply_cooldown()
+	..()
+	return
+
+
+/datum/action/xeno_action/activable/rooting_slash/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/X = owner
+	if (!action_cooldown_check())
+		return
+
+	if (!X.check_state())
+		return
+
+	if (!check_and_use_plasma_owner())
+		return
+
+	if (!isXenoOrHuman(A) || X.can_not_harm(A))
+		to_chat(X, SPAN_XENODANGER("You must target a hostile!"))
+		return
+
+	var/mob/living/carbon/H = A
+
+	var/dist = get_dist(X, H)
+
+	if (dist > range)
+		to_chat(X, SPAN_WARNING("[H] is too far away!"))
+		return
+
+	if (dist > 1)
+		var/turf/targetTurf = get_step(X, get_dir(X, H))
+		if (targetTurf.density)
+			to_chat(X, SPAN_WARNING("You can't attack through [targetTurf]!"))
+			return
+		else
+			for (var/atom/I in targetTurf)
+				if (I.density && !I.throwpass && !istype(I, /obj/structure/barricade) && !istype(I, /mob/living))
+					to_chat(X, SPAN_WARNING("You can't attack through [I]!"))
+					return
+
+
+	if (H.stat == DEAD)
+		to_chat(X, SPAN_XENODANGER("[H] is dead, why would you want to touch it?"))
+		return
+
+	if (X.mutation_type == ROYAL_NORMAL)
+		var/datum/behavior_delegate/royal/BD = X.behavior_delegate
+		if (!istype(BD))
+			return
+
+
+
+	// Flick overlay and play sound
+	X.animation_attack_on(A, 10)
+	var/S = pick('sound/weapons/punch1.ogg','sound/weapons/punch2.ogg','sound/weapons/punch3.ogg','sound/weapons/punch4.ogg')
+	playsound(H,S, 50, 1)
+
+	var/root_duration = 1 SECONDS
+	var/damage = 15
+	X.visible_message(SPAN_XENODANGER("[X] extends with its claws and smashes [A], pinning them to the ground dealing damage!"), SPAN_XENOHIGHDANGER("You extend your claws and smash [A], pinning them to the ground and dealing damage!"))
+
+	H.frozen = TRUE
+	H.update_canmove()
+	H.apply_armoured_damage(damage, ARMOR_MELEE, BRUTE, "chest", 15)
+
+	if (ishuman(H))
+		var/mob/living/carbon/human/Hu = H
+		Hu.update_xeno_hostile_hud()
+
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/unroot_human, H), get_xeno_stun_duration(H, root_duration))
+		to_chat(H, SPAN_XENOHIGHDANGER("[X] has pinned you to the ground! You cannot move!"))
+
+	apply_cooldown()
+	..()
+	return
+
+/datum/action/xeno_action/activable/cleave/proc/remove_bufff()
```

Comment:
extra f on the proc name

Reply:
it conflicts with another var, i'll change it to remove_enhancement or some shit